middleware/src/middleware.go

```diff
@@ -0,0 +1,105 @@
+package middleware
+
+import (
+	"time"
+	//"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/rpcclient"
+	//"github.com/btcsuite/btcutil"
+	"github.com/fiatjaf/lightningd-gjson-rpc"
+	//"github.com/tidwall/gjson"
+	"github.com/digitalbitbox/bitbox-base/middleware/util/logging"
+	"github.com/sirupsen/logrus"
+)
+
+type MiddlewareInfoStruct struct {
+	Blocks         int64   `json:"blocks"`
+	Difficulty     float64 `json:"difficulty"`
+	LightningAlias string  `json:"alias"`
+}
+
+type Middleware struct {
+	info                                               MiddlewareInfoStruct
+	events                                             chan *MiddlewareInfoStruct
+	bitcoinRpcUser, bitcoinRpcPassword, bitcoinRpcPort string
+	lightningRpcPath                                   string
+	log                                                *logrus.Entry
+}
+
+// NewMiddleware creates a new instance of the middleware
+func NewMiddleware(bitcoinRpcUser, bitcoinRpcPassword, bitcoinRpcPort, lightningRpcPath string) *Middleware {
+	middleware := &Middleware{
+		events:             make(chan *MiddlewareInfoStruct),
+		bitcoinRpcUser:     bitcoinRpcUser,
+		bitcoinRpcPassword: bitcoinRpcPassword,
+		bitcoinRpcPort:     bitcoinRpcPort,
+		lightningRpcPath:   lightningRpcPath,
+		log:                logging.Get().WithGroup("middleware"),
+	}
+	return middleware
+}
+
+func (middleware *Middleware) testBitcoinRPC(bitcoinRpcUser, bitcoinRpcPassword, bitcoinRpcPort string) MiddlewareInfoStruct {
```

Comment:
I think I understand what's going on after reading further in this file, but I was initially surprised to see a method called `testFoo()` in a file not called `bar_test.go`..

Perhaps a name like `checkBitcoinRPCConnection()` or similar would be more clear?

(And elsewhere.)

Reply:
Renamed.